flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -875,7 +879,13 @@ private static void ensureRocksDBIsLoaded(String tempDirectory) throws IOExcepti
 						//  instances of the same JNI library being loaded in different class loaders, but
 						//  apparently not when coming from the same file path, so there we go)
 
-						final File rocksLibFolder = new File(tempDirParent, "rocksdb-lib-" + new AbstractID());
+						rocksLibFolder = new File(tempDirParent, "rocksdb-lib-" + new AbstractID());
+						// make sure the temp path exists
+						LOG.debug("Attempting to create RocksDB native library folder {}", rocksLibFolder);
+
+						// noinspection ResultOfMethodCallIgnored
+						rocksLibFolder.mkdirs();
+						rocksLibFolder.deleteOnExit();
```

Comment:
The directory should be deleted anyways on shutdown, because the TaskManagers have a shutdown hook to clean up their temp directories.

I would suggest to drop `deleteOnExit()`, because as far as I know it can also be problematic and lead to memory leaks.